pkg/services/sqlstore/migrations/playlist_mig.go

```diff
@@ -44,3 +34,32 @@ func addPlaylistMigrations(mg *Migrator) {
 		{Name: "title", Type: DB_Text, Nullable: false},
 	}))
 }
+
+func addPlaylistUIDMigration(mg *Migrator) {
+	// Replacing auto-incremented playlistIDs with string UIDs
+	mg.AddMigration("Add UID column to playlist", NewAddColumnMigration(playlistV2(), &Column{
+		Name: "uid", Type: DB_NVarchar, Length: 80, Nullable: false, Default: "0",
+	}))
+
+	// copy the (string representation of) existing IDs into the new uid column.
+	mg.AddMigration("Update uid column values in playlist", NewRawSQLMigration("").
+		SQLite("UPDATE playlist SET uid=printf('%d',id);").
+		Postgres("UPDATE playlist SET uid=id::text;").
+		Mysql("UPDATE playlist SET uid=id;"))
```

Comment:
Normally I'd recommend that a DDL statement like this is written into batches to avoid long-running exclusive locks on the database, but looking at our usage of playlists for out grafana cloud customers, [I can see the heaviest user only has 30 playlists](https://ops.grafana.net/goto/EHofq597k?orgId=1). Therefore I don't think the impact of updating all these at once would have negative impact from a cloud perspective.